lib/mox.ex

```diff
@@ -812,7 +817,7 @@ defmodule Mox do
   def __dispatch__(mock, name, arity, args) do
     all_callers = [self() | caller_pids()]
 
-    case Mox.Server.fetch_fun_to_dispatch(all_callers, {mock, name, arity}) do
+    case Mox.Server.fetch_fun_to_dispatch(all_callers, {mock, name, arity}, args) do
```

Comment:
This means we are copying all argument data to a separate process. :(

Reply:
Right, is that an issue in tests? We would do the same when using the `send/2` trick to send calls to the parent process.

Any ideas on how to do this without copying the args?

Reply:
the problem is that this enables this for everything by default. We would need to store it somewhere if we want to collect or not.